contrib/test_mad.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+set -exE -o pipefail
+
+cd "$BUILD_SOURCESDIRECTORY"
+
+run_mad_server() {
+    build_ucx
+    setup
+    srv_stop
+    funcname
+    export HCA="$HCA"
+    sudo -E bash -c 'envsubst < "buildlib/tools/ucx_perftest.template" \
+        > /etc/systemd/system/ucx_perftest.service'
+    sudo systemctl daemon-reload
+    sudo systemctl start ucx_perftest
+    sudo systemctl status ucx_perftest
+}
+
+build_ucx() {
+    funcname
+    ./autogen.sh
+    ./contrib/configure-release \
+        --prefix="$PWD"/install \
+        --with-mad \
+        --without-valgrind \
+        --without-go \
+        --without-java
+    make -s -j"$(nproc)"
+    make install
+}
+
+setup() {
+    funcname
+    sudo chmod 777 /dev/infiniband/umad*
+}
+
+set_vars() {
+    set +x
+    HCA=${HCA/:/ } # Replace ':' with space
+    # shellcheck disable=SC2086
+    LID=$(ibstat $HCA | grep Base | awk '{print $NF}')
+    # shellcheck disable=SC2086
+    GUID=$(ibstat $HCA | grep GUID | awk '{print $NF}')
+    echo "LID: $LID"
+    echo "GUID: $GUID"
+    echo "##vso[task.setvariable variable=LID;isOutput=true]$LID"
+    echo "##vso[task.setvariable variable=GUID;isOutput=true]$GUID"
+}
+
+run_mad_test_lid() {
+    build_ucx
```

Comment:
why we build ucx as part of this test? it should be separated function, what if someone decide to run only guid tests it will not work

Reply:
A separate “Setup” job has been created, including UCX build and system setup.